Client/Ecosia/UI/MultiplyImpact/MultiplyImpact.swift

```diff
@@ -506,18 +507,17 @@ final class MultiplyImpact: UIViewController, NotificationThemeable {
     }
 
     private func updateInviteLink() {
-        guard let inviteLink = inviteLink else {
-            copyLink?.text = "-"
-            referrals.refresh(createCode: true) {[weak self] error in
-                if let error = error {
-                    self?.showObtainingCode(error)
-                } else {
-                    self?.updateInviteLink()
-                }
+        copyLink?.text = inviteLink ?? inviteLinkPlaceholder
+    }
+    
+    private func refreshReferrals() {
+        referrals.refresh(force: true, createCode: true) { [weak self] error in
+            guard error == nil else {
+                self?.showObtainingCode(error!)
+                return
             }
-            return
+            self?.updateInviteLink()
```

Comment:
Also believe you need to update the UI for the counter and not only the invite link after the refresh happens. If you update from main with the changes from #568, this should be only updating `referralImpactRowView.info`.